crates/vhost/src/vhost_user/message.rs

```diff
@@ -528,44 +529,36 @@ pub type VhostUserMemoryPayload = Vec<VhostUserMemoryRegion>;
 pub struct VhostUserSingleMemoryRegion {
     /// Padding for correct alignment
     padding: u64,
-    /// Guest physical address of the memory region.
-    pub guest_phys_addr: u64,
-    /// Size of the memory region.
-    pub memory_size: u64,
-    /// Virtual address in the current process.
-    pub user_addr: u64,
-    /// Offset where region starts in the mapped memory.
-    pub mmap_offset: u64,
+    /// General memory region
+    region: VhostUserMemoryRegion,
+}
+
+impl Deref for VhostUserSingleMemoryRegion {
+    type Target = VhostUserMemoryRegion;
+
+    fn deref(&self) -> &VhostUserMemoryRegion {
+        &self.region
+    }
 }
 
 impl VhostUserSingleMemoryRegion {
     /// Create a new instance.
     pub fn new(guest_phys_addr: u64, memory_size: u64, user_addr: u64, mmap_offset: u64) -> Self {
         VhostUserSingleMemoryRegion {
             padding: 0,
-            guest_phys_addr,
-            memory_size,
-            user_addr,
-            mmap_offset,
+            region: VhostUserMemoryRegion::new(
+                guest_phys_addr,
+                memory_size,
+                user_addr,
+                mmap_offset,
+            ),
         }
     }
 }
 
 // SAFETY: Safe because all fields of VhostUserSingleMemoryRegion are POD.
 unsafe impl ByteValued for VhostUserSingleMemoryRegion {}
-
-impl VhostUserMsgValidator for VhostUserSingleMemoryRegion {
-    fn is_valid(&self) -> bool {
-        if self.memory_size == 0
-            || self.guest_phys_addr.checked_add(self.memory_size).is_none()
-            || self.user_addr.checked_add(self.memory_size).is_none()
-            || self.mmap_offset.checked_add(self.memory_size).is_none()
-        {
-            return false;
-        }
-        true
-    }
-}
+impl VhostUserMsgValidator for VhostUserSingleMemoryRegion {}
```

Comment:
This can be just dropped with the Deref impl in place, or am I missing something?

Reply:
That gives:

```
the trait `message::VhostUserMsgValidator` is not implemented for `message::VhostUserSingleMemoryRegion``
```

Reply:
Hm... I did not see where it was used as type, but looks like I commented on unrelated stuff anyway.